src/mlpack/methods/ann/layer/batch_norm_impl.hpp

```diff
@@ -106,7 +108,6 @@ template<typename eT>
 void BatchNorm<InputDataType, OutputDataType>::Backward(
     const arma::Mat<eT>&& input, arma::Mat<eT>&& gy, arma::Mat<eT>&& g)
 {
-  const arma::mat inputMean = input.each_col() - mean;
```

Comment:
I see, we were not using the processed output. I remember facing such an issue in a part implementation of `ELU` as well. Nice catch.